src/AWS.Deploy.CLI/Commands/TypeHints/ExistingSubnetsCommand.cs

```diff
@@ -0,0 +1,52 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Amazon.EC2.Model;
+using AWS.Deploy.Common;
+using AWS.Deploy.Common.Recipes;
+using AWS.Deploy.Common.TypeHintData;
+using AWS.Deploy.Orchestration.Data;
+
+namespace AWS.Deploy.CLI.Commands.TypeHints
+{
+    public class ExistingSubnetsCommand : ITypeHintCommand
+    {
+        private readonly IAWSResourceQueryer _awsResourceQueryer;
+        private readonly IConsoleUtilities _consoleUtilities;
+
+        public ExistingSubnetsCommand(IAWSResourceQueryer awsResourceQueryer, IConsoleUtilities consoleUtilities)
+        {
+            _awsResourceQueryer = awsResourceQueryer;
+            _consoleUtilities = consoleUtilities;
+        }
+
+        private async Task<List<Subnet>> GetData()
+        {
+            return await _awsResourceQueryer.DescribeSubnets();
```

Comment:
After setting the `ParentSettingId` in the recipe as desribed below. Fetch the VpcID using `recommendation.GetOptionSetting(optionSetting.ParentSettingId)` like it is done in the Beanstalk Environment type hint. https://github.com/aws/aws-dotnet-deploy/blob/main/src/AWS.Deploy.CLI/Commands/TypeHints/BeanstalkEnvironmentCommand.cs#L30

Be sure to handle `ParentSettingId` not existing.